File: Phidget Code/src/Models/In/Goal.java
package Models.In;

import Events.EventType;
import Events.PassageEvent;
import Events.VibrationEvent;
import Models.BlackBox.BlackBox;
import com.phidgets.InterfaceKitPhidget;
import com.phidgets.PhidgetException;
import com.phidgets.event.SensorChangeEvent;
import com.phidgets.event.SensorChangeListener;

/**
 * Created by bri_e on 07-03-17.
 */
public class Goal {

    private static int MILI_INTERVAL = 100;

    private int indexIRSensor;
    private int indexVibrationSensor;
    private int goal;
    private double lastGoal;
    private BlackBox blackBox;
    private InterfaceKitPhidget interfaceKitPhidget;

    public Goal(InterfaceKitPhidget ifk, int indexPassage, int indexVibration, BlackBox bbx) throws PhidgetException {
        blackBox = bbx;
        interfaceKitPhidget = ifk;
        indexIRSensor = indexPassage;
        indexVibrationSensor = indexVibration;
        goal = 0;
        lastGoal=0;
        ifk.addSensorChangeListener(new SensorChangeListener() {
            @Override
            public void sensorChanged(SensorChangeEvent sensorChangeEvent) {
                switch(sensorChangeEvent.getIndex()) {
                    case 3 :
                        //System.out.println("Passage !");
                        try {
                            if (ifk.getSensorValue(indexIRSensor) < 100) {
                                if ((blackBox.noEvent(EventType.PASSAGE_EVENT))
                                        || ((System.currentTimeMillis() - blackBox.getLast(EventType.PASSAGE_EVENT).removeLast().getTime()) > MILI_INTERVAL)) {
                                    PassageEvent passageEvent = new PassageEvent(System.currentTimeMillis());
                                    blackBox.processElement(passageEvent);
                                }
                            }
                        } catch (PhidgetException e) {
                            System.out.println("Exception à la lecture du phidget de proximité : " + e);
                        }
                        break;
                    case 4:
                        //System.out.println("Vibration !");
                        if((blackBox.noEvent(EventType.VIBRATION_EVENT))
                                ||((System.currentTimeMillis() - blackBox.getLast(EventType.VIBRATION_EVENT).removeLast().getTime()) > MILI_INTERVAL)) {

                            try {
                                if (ifk.getSensorValue(indexVibrationSensor) > 500) {
                                    VibrationEvent vibrationEvent = new VibrationEvent(System.currentTimeMillis());
                                    blackBox.processElement(vibrationEvent);
                                }
                            } catch (PhidgetException e) {
                                System.out.println("Exception à la lecture du phidget de vibration : " + e);
                            }
                        }
                        break;
                    default:
                        break;

                }
            }
        });

    }

    public int getGoal(){
        return goal;
    }

    public double getLastGoal(){
        return lastGoal;
    }

    public void incrementGoal(double time){
        goal++;
        lastGoal = time;
        System.out.println("Goal ! : " + this.getGoal() + " at " + lastGoal);
    }




}

File: PhidgetScalaCode/src/Models/In/Goal.scala
package Models.In

import BlackBox.BlackBox
import Events.PassageEvent
import com.phidgets.InterfaceKitPhidget
import com.phidgets.event.{SensorChangeEvent, SensorChangeListener}

/**
  * Created by bri_e on 20-04-17.
  * This object represent a goal, we listen to both passage event and vibration event.
  * With a calculation on the time lapse between both event in a correct order we will determinate if there is a goal or not
  */
class Goal(interfaceKitPhidget: InterfaceKitPhidget, indexPassage: Int, indexVibration: Int, blackBox: BlackBox) {

  var goal:Int = 0
  var lastGoal:Double = 0.0

  interfaceKitPhidget.addSensorChangeListener(new SensorChangeListener {
    override def sensorChanged(sensorChangeEvent: SensorChangeEvent): Unit = {
      sensorChangeEvent.getIndex match {
        case indexPassage =>
          if (interfaceKitPhidget.getSensorValue(indexPassage) < 100){
            //Todo check if noEvent + timout  if ok =>  new passage event + proceed it
          }
        case indexVibration => {
          if (interfaceKitPhidget.getSensorValue(indexPassage) > 500){
            //Todo check if noEvent + timout if ok => new vibration event + proceed it
          }
        }
      }
    }
  })

  def getGoal:Int = goal
  def getLastGoal:Double = lastGoal
  def incrementGoal(time:Double):Unit = {
    goal += 1
    lastGoal = time
  }
}
